pandas/core/dtypes/cast.py

```diff
@@ -2212,4 +2251,11 @@ def can_hold_element(dtype: np.dtype, element: Any) -> bool:
     elif dtype == object:
         return True
 
+    elif dtype.kind == "S":
```

Comment:
do we ever have kind == 'U"?

Reply:
we should *never* have this actually (I think we are deprecating)?

Reply:
my understanding is that this shouldnt be reached, but de facto it is and i decided fixing that was orthogonal to this PR